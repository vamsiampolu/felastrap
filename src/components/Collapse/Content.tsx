import { CSSProperties } from "fela";
import { useContext } from "react";
import { useFela } from "react-fela";
import { Portal } from "../Portal";
import { StyleProps, type DivProps } from "../common";
import { CollapseContext, CollapseContextValue } from "./CollapseContext";

export type CollapseContentProps = DivProps & { detached?: boolean };
type Height = CSSProperties["height"];

const rule = ({ height }: StyleProps<CollapseContextValue>) => ({
  overflow: "hidden",
  width: "auto",
  height: height as Height,
  transition: "height 0.35s ease",
  "@media (prefers-reduced-motion: reduce)": {
    transition: "none",
  },
});

export const Content = ({
  id = "",
  className = "",
  detached = false,
  children,
  style = {},
}: CollapseContentProps) => {
  const { contentRef, height } = useContext(CollapseContext);
  const { css } = useFela({ height });

  const cx = `${css([rule])} collapse ${className}`;
  if (detached) {
    return (
      <Portal
        wrapperId={`wrapper-collapse-content-${id}`}
        style={style as CSSStyleDeclaration}
      >
        <div data-testid="collapse" className={cx}>
          <div data-testid="collapse-content" ref={contentRef} id={id}>
            <div data-testid="collapse-container" className="content-container">
              {children}
            </div>
          </div>
        </div>
      </Portal>
    );
  }

  return (
    <div data-testid="collapse" className={cx}>
      <div data-testid="collapse-content" ref={contentRef} id={id}>
        <div data-testid="collapse-container" className="content-container">
          {children}
        </div>
      </div>
    </div>
  );
};
