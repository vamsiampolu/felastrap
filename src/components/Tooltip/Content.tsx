import { ForwardedRef, forwardRef } from "react";
import { DivProps } from "../common";
import { CSSProperties } from "fela";
import { useFela } from "react-fela";

export type ContentProps = DivProps & {
  open?: boolean;
  transitionStyles: CSSProperties;
  tooltip: string;
};

const rule = () => ({
  zIndex: 1080,
  display: "block",
  fontFamily:
    'system-ui, -apple-system, "Segoe UI", Roboto, "Helvetica Neue", "Noto Sans", "Liberation Sans", Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"',
  fontStyle: "normal",
  fontWeight: 400,
  lineHeight: 1.5,
  textAlign: "start",
  textDecoration: "none",
  textShadow: "none",
  textTransform: "none",
  letterSpacing: "normal",
  wordBreak: "normal",
  whiteSpace: "normal",
  wordSpacing: "normal",
  lineBreak: "auto",
  fontSize: "0.875rem",
  wordWrap: "break-word",
  opacity: 0.9,
});

const innerRule = () => ({
  maxWidth: "200px",
  padding: "0.25rem 0.5rem",
  color: "#fff",
  textAlign: "center",
  backgroundColor: "#000",
  borderRadius: "0.375rem",
});

export const ContentComponent = (
  {
    tooltip,
    children,
    style,
    className = "",
    transitionStyles,
    open = false,
    ...props
  }: ContentProps,
  ref: ForwardedRef<HTMLDivElement>,
) => {
  const { css } = useFela();

  if (!open) {
    return null;
  }

  const cx = `${css(rule)} ${className}`;
  const floatingClassName = `${css(innerRule)} floating`;

  return (
    <div
      data-testid="tooltip-content"
      className={cx}
      ref={ref}
      style={style}
      {...props}
    >
      <div
        data-testid="tooltip-content-inner"
        style={transitionStyles}
        className={floatingClassName}
      >
        {tooltip}
      </div>
      {children}
    </div>
  );
};

export const Content = forwardRef(ContentComponent);
