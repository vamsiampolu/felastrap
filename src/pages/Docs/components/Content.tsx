import { PropsWithChildren, ReactNode } from "react";
import { useFela } from "react-fela";
import { Intro } from "./Intro";
import { TableOfContents } from "./TableOfContents/TableOfContents";

const rule = () => ({
  gridArea: "main",
  display: "grid",

  order: 1,
  overflowWrap: "break-word",
  textWrap: "wrap",
  "@media (min-width: 768px)": {
    gridTemplateAreas: '"intro" "toc" "content"',
    gridTemplateRows: "auto auto 1fr",
    gap: "inherit",
  },
  "@media (min-width: 992px)": {
    gridTemplateAreas: `
    "intro   toc"
    "content toc"
`
      .split("\n")
      .map((l) => l.trim())
      .join("\n"),
    gridTemplateRows: "auto 1fr",
    gridTemplateColumns: "4fr 1fr",
  },
});

export type ContentProps = PropsWithChildren<{
  className?: string;
  title: ReactNode;
  description: ReactNode;
  toc: ReactNode;
}>;

export const Content = ({
  className = "",
  title,
  description,
  toc,
  children,
}: ContentProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} main ${className}`.trim();

  return (
    <main className={cx}>
      <Intro title={title} description={description} />
      <TableOfContents>{toc}</TableOfContents>
      {children}
    </main>
  );
};
