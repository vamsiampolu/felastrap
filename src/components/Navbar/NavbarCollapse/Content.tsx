import { useFela } from "react-fela";
import { Content as CollapseContent } from "../../Collapse";
import type { CollapseContentProps } from "../../Collapse/Content";
import { mediaQueryByName } from "../../Flex/mediaQuery";
import { NavbarSize, sizeBreakpointMapping } from "../utils";
import { NavbarVariant } from "../Navbar";
import { StyleProps } from "../../common";

const rule = () => ({
  flexBasis: "100%",
  flexGrow: 1,
  alignItems: "center",
});

const expandSizeRule = ({ size, variant }: StyleProps<ContentProps>) => {
  if (variant === "expand") {
    const breakpoint = sizeBreakpointMapping[size as NavbarSize];
    return {
      [mediaQueryByName[breakpoint]]: {
        display: "flex !important",
        flexBasis: "auto",
        flexGrow: 1,
        alignItems: "center",
      },
    };
  }

  return {};
};

export type ContentProps = CollapseContentProps & {
  size?: NavbarSize;
  variant?: NavbarVariant;
};

export const Content = ({
  className = "",
  size = "medium",
  variant = "expand",
  ...props
}: ContentProps) => {
  const { css } = useFela({ size, variant });
  const cx = `${css([
    rule,
    expandSizeRule,
  ])} navbar-collapse ${className}`.trim();
  return <CollapseContent className={cx} {...props} />;
};
