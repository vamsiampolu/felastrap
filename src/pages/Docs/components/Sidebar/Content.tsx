import { useFela } from "react-fela";
import { DivProps } from "../../../../components/common";

const rule = () => ({
  flexGrow: 1,
  padding: "1rem 1rem",
  overflowY: "auto",
  "@media (min-width: 992px)": {
    marginTop: "1.5rem",
    display: "flex",
    flexGrow: "0",
    padding: "0",
    overflowY: "visible",
    backgroundColor: "transparent !important",
  },
});

export const Content = ({ children, className = "" }: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} offcanvas-body ${className}`.trim();
  return <div className={cx}>{children}</div>;
};
