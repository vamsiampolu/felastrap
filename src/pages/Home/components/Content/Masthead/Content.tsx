import { useFela } from "react-fela";
import { Column } from "../../../../../components/Flex";
import { Link } from "../../../../../components/Link/Link";
import { Header } from "./Header";
import { HeroDetails } from "./HeroDetails";
import { HeroImage } from "./HeroImage";
import { HeroText } from "./HeroText";
import { Lead } from "./Lead";

const rule = () => ({
  color: "rgba(33, 37, 41, 0.75)",
  marginBottom: 0,
  marginTop: 0,
});

const linkRule = () => ({
  color: "rgba(108,117,125, 1)",
  textDecorationColor: "rgba(108,117,125, 1)",
});

const linkWhiteSpaceRule = () => ({
  whiteSpace: "nowrap !important",
});

export const HeroSubtext = () => {
  const { css } = useFela();
  const cx = `${css(rule)} hero-subtext`;
  return (
    <p className={cx}>
      Currently <strong>v5.3.2</strong>
      <span
        className={css({ paddingRight: "0.25rem", paddingLeft: "0.25rem" })}
      >
        .
      </span>
      <Link
        className={css(linkRule)}
        href="/docs/5.3/getting-started/download/"
      >
        Download
      </Link>
      <span
        className={css({ paddingRight: "0.25rem", paddingLeft: "0.25rem" })}
      >
        .
      </span>
      <Link
        className={css([linkRule, linkWhiteSpaceRule])}
        href="/docs/versions/"
      >
        Download
      </Link>
    </p>
  );
};

export const Content = () => {
  const { css } = useFela();
  const columnCx = css({
    marginRight: "auto",
    marginLeft: "auto",
    textAlign: "center",
  });

  return (
    <Column md={8} className={columnCx}>
      <Header
        title="New in v5.3"
        href="https://blog.getbootstrap.com/"
        text="Color mode support, expanded color palette, and more!"
      />
      <HeroImage />
      <HeroText>Build fast, responsive sites with Bootstrap</HeroText>
      <Lead>
        Powerful, extensible, and feature-packed frontend toolkit. Build and
        customize with Sass, utilize prebuilt grid system and components, and
        bring projects to life with powerful JavaScript plugins.
      </Lead>
      <HeroDetails />
      <HeroSubtext />
    </Column>
  );
};
