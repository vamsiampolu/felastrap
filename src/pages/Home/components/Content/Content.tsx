import { Followup } from "./Followup/Followup";
import { Masthead } from "./Masthead/Masthead";

export const Content = () => {
  return (
    <main>
      <Masthead />
      <Followup />
    </main>
  );
};
