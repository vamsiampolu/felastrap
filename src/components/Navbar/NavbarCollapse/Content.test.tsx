import { CSSProperties, IRenderer } from "fela";
import { Content, ContentProps } from "./Content";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../../test-helpers";
import { useRef } from "react";
import {
  CollapseContext,
  CollapseContextValue,
} from "../../Collapse/CollapseContext";
import { render, screen } from "@testing-library/react";

const ContentExample = ({
  isOpen,
  height,
  ...contentProps
}: { isOpen?: boolean; height?: CSSProperties["height"] } & ContentProps) => {
  const contentRef = useRef(null);
  const toggleCollapse = () => undefined;
  const value: CollapseContextValue = {
    contentRef,
    toggleCollapse,
    height: height ?? 0,
    collapsed: !(isOpen ?? false),
    isOpen: isOpen ?? false,
  };

  return (
    <CollapseContext.Provider value={value}>
      <Content {...contentProps} />
    </CollapseContext.Provider>
  );
};

describe("components/Navbar/NavbarCollapse/Content", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a navbar collapse content component", async () => {
    const component = (
      <ContentExample isOpen={false}>
        <div data-testid="hello-there">Hello, there</div>
      </ContentExample>
    );
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("collapse")).toBeInTheDocument();
    expect(screen.getByTestId("hello-there")).toBeInTheDocument();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render an open navbar collapse content component", async () => {
    const component = (
      <ContentExample isOpen={true}>
        <div data-testid="hello-there">Hello, there</div>
      </ContentExample>
    );
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("collapse")).toBeInTheDocument();
    expect(screen.getByTestId("hello-there")).toBeInTheDocument();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a navbar collapse content component with variant sticky", async () => {
    const component = (
      <ContentExample isOpen={false} variant="sticky">
        <div data-testid="hello-there">Hello, there</div>
      </ContentExample>
    );
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("collapse")).toBeInTheDocument();
    expect(screen.getByTestId("hello-there")).toBeInTheDocument();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a navbar collapse content component with variant expand and size small", async () => {
    const component = (
      <ContentExample isOpen={false} variant="expand" size="small">
        <div data-testid="hello-there">Hello, there</div>
      </ContentExample>
    );
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("collapse")).toBeInTheDocument();
    expect(screen.getByTestId("hello-there")).toBeInTheDocument();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a navbar collapse content component with variant fixed and size small", async () => {
    const component = (
      <ContentExample isOpen={false} variant="fixed" size="small">
        <div data-testid="hello-there">Hello, there</div>
      </ContentExample>
    );
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("collapse")).toBeInTheDocument();
    expect(screen.getByTestId("hello-there")).toBeInTheDocument();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
