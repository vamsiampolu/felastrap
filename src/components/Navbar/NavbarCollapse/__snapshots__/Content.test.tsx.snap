// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navbar/NavbarCollapse/Content should render a navbar collapse content component 1`] = `
".a {
  flex-basis: 100%;
}
.b {
  flex-grow: 1;
}
.c {
  align-items: center;
}
.h {
  width: auto;
}
.i {
  height: 0;
}
.n {
  transition-property: height;
}
.o {
  transition-duration: 0.35s;
}
.p {
  transition-delay: ease;
}
.q {
  transition-timing-function: ease;
}
.r {
  overflow-x: hidden;
}
.s {
  overflow-y: hidden;
}
@media (min-width: 768px) {
  .d {
    display: flex !important;
  }
  .e {
    flex-basis: auto;
  }
  .f {
    flex-grow: 1;
  }
  .g {
    align-items: center;
  }
}
@media (prefers-reduced-motion: reduce) {
  .j {
    transition-property: none;
  }
  .k {
    transition-duration: 0s;
  }
  .l {
    transition-delay: 0s;
  }
  .m {
    transition-timing-function: ease;
  }
}


<div
  data-testid="collapse"
  class="h i j k l m n o p q r s collapse a b c d e f g navbar-collapse"
>
  <div data-testid="collapse-content" id="">
    <div data-testid="collapse-container" class="content-container">
      <div data-testid="hello-there">Hello, there</div>
    </div>
  </div>
</div>
;
"
`;

exports[`components/Navbar/NavbarCollapse/Content should render a navbar collapse content component with variant expand and size small 1`] = `
".a {
  flex-basis: 100%;
}
.b {
  flex-grow: 1;
}
.c {
  align-items: center;
}
.h {
  width: auto;
}
.i {
  height: 0;
}
.n {
  transition-property: height;
}
.o {
  transition-duration: 0.35s;
}
.p {
  transition-delay: ease;
}
.q {
  transition-timing-function: ease;
}
.r {
  overflow-x: hidden;
}
.s {
  overflow-y: hidden;
}
@media (min-width: 576px) {
  .d {
    display: flex !important;
  }
  .e {
    flex-basis: auto;
  }
  .f {
    flex-grow: 1;
  }
  .g {
    align-items: center;
  }
}
@media (prefers-reduced-motion: reduce) {
  .j {
    transition-property: none;
  }
  .k {
    transition-duration: 0s;
  }
  .l {
    transition-delay: 0s;
  }
  .m {
    transition-timing-function: ease;
  }
}


<div
  data-testid="collapse"
  class="h i j k l m n o p q r s collapse a b c d e f g navbar-collapse"
>
  <div data-testid="collapse-content" id="">
    <div data-testid="collapse-container" class="content-container">
      <div data-testid="hello-there">Hello, there</div>
    </div>
  </div>
</div>
;
"
`;

exports[`components/Navbar/NavbarCollapse/Content should render a navbar collapse content component with variant fixed and size small 1`] = `
".a {
  flex-basis: 100%;
}
.b {
  flex-grow: 1;
}
.c {
  align-items: center;
}
.d {
  width: auto;
}
.e {
  height: 0;
}
.j {
  transition-property: height;
}
.k {
  transition-duration: 0.35s;
}
.l {
  transition-delay: ease;
}
.m {
  transition-timing-function: ease;
}
.n {
  overflow-x: hidden;
}
.o {
  overflow-y: hidden;
}
@media (prefers-reduced-motion: reduce) {
  .f {
    transition-property: none;
  }
  .g {
    transition-duration: 0s;
  }
  .h {
    transition-delay: 0s;
  }
  .i {
    transition-timing-function: ease;
  }
}


<div
  data-testid="collapse"
  class="d e f g h i j k l m n o collapse a b c navbar-collapse"
>
  <div data-testid="collapse-content" id="">
    <div data-testid="collapse-container" class="content-container">
      <div data-testid="hello-there">Hello, there</div>
    </div>
  </div>
</div>
;
"
`;

exports[`components/Navbar/NavbarCollapse/Content should render a navbar collapse content component with variant sticky 1`] = `
".a {
  flex-basis: 100%;
}
.b {
  flex-grow: 1;
}
.c {
  align-items: center;
}
.d {
  width: auto;
}
.e {
  height: 0;
}
.j {
  transition-property: height;
}
.k {
  transition-duration: 0.35s;
}
.l {
  transition-delay: ease;
}
.m {
  transition-timing-function: ease;
}
.n {
  overflow-x: hidden;
}
.o {
  overflow-y: hidden;
}
@media (prefers-reduced-motion: reduce) {
  .f {
    transition-property: none;
  }
  .g {
    transition-duration: 0s;
  }
  .h {
    transition-delay: 0s;
  }
  .i {
    transition-timing-function: ease;
  }
}


<div
  data-testid="collapse"
  class="d e f g h i j k l m n o collapse a b c navbar-collapse"
>
  <div data-testid="collapse-content" id="">
    <div data-testid="collapse-container" class="content-container">
      <div data-testid="hello-there">Hello, there</div>
    </div>
  </div>
</div>
;
"
`;

exports[`components/Navbar/NavbarCollapse/Content should render an open navbar collapse content component 1`] = `
".a {
  flex-basis: 100%;
}
.b {
  flex-grow: 1;
}
.c {
  align-items: center;
}
.h {
  width: auto;
}
.i {
  height: 0;
}
.n {
  transition-property: height;
}
.o {
  transition-duration: 0.35s;
}
.p {
  transition-delay: ease;
}
.q {
  transition-timing-function: ease;
}
.r {
  overflow-x: hidden;
}
.s {
  overflow-y: hidden;
}
@media (min-width: 768px) {
  .d {
    display: flex !important;
  }
  .e {
    flex-basis: auto;
  }
  .f {
    flex-grow: 1;
  }
  .g {
    align-items: center;
  }
}
@media (prefers-reduced-motion: reduce) {
  .j {
    transition-property: none;
  }
  .k {
    transition-duration: 0s;
  }
  .l {
    transition-delay: 0s;
  }
  .m {
    transition-timing-function: ease;
  }
}


<div
  data-testid="collapse"
  class="h i j k l m n o p q r s collapse a b c d e f g navbar-collapse"
>
  <div data-testid="collapse-content" id="">
    <div data-testid="collapse-container" class="content-container">
      <div data-testid="hello-there">Hello, there</div>
    </div>
  </div>
</div>
;
"
`;
